RosNet/RosReader/RosBagReader.cs

```diff
@@ -61,8 +83,257 @@ public static RosBag Read(string path)
+                FieldValue fieldValue = ReadField(reader);
+                headerFields.Add(fieldValue.Name, fieldValue);
+
+                if (headerFields.ContainsKey("op"))
+                {
+                    hasAllFields = true;
+                    foreach (string headerField in HeaderFieldsByOp[(int)headerFields["op"].Value.First()])
+                    {
+                        if (!headerFields.ContainsKey(headerField))
+                        {
+                            hasAllFields = false;
+                        }
+                    }
+                }
+            }
+            return headerFields;
+        }
+
+        /// <summary>
+        /// Reads a field
+        /// </summary>
+        /// <returns>FieldValue containing name, datatype and value of field</returns>
+        private static FieldValue ReadField(BinaryReader reader)
+        {
+            int fieldLen = reader.ReadInt32();
+            string fieldName = ReadName(reader);
+
+            PrimitiveType dataType;
+            byte[] fieldValue;
+            switch (fieldName)
+            {
+                case "index_pos":
+                case "time":
+                case "chunk_pos":
+                case "start_time":
+                case "end_time":
+                    dataType = PrimitiveType.INT64;
+                    fieldValue = reader.ReadBytes(8);
+                    break;
+                case "conn_count":
+                case "chunk_count":
+                case "size":
+                case "conn":
+                case "ver":
+                case "count":
+                case "offset":
+                    dataType = PrimitiveType.INT32;
+                    fieldValue = reader.ReadBytes(4);
+                    break;
+                case "op":
+                    dataType = PrimitiveType.INT8;
+                    fieldValue = reader.ReadBytes(1);
+                    break;
+                case "compression":
+                    dataType = PrimitiveType.STRING;
+                    char firstChar = reader.ReadChar();
+                    if (firstChar.Equals('n'))
+                    {
+                        reader.ReadChars(3);
+                        fieldValue = Encoding.ASCII.GetBytes("none");
+                    }
+                    else
+                    {
+                        reader.ReadChars(2);
+                        fieldValue = Encoding.ASCII.GetBytes("bz2");
+                    }
+                    break;
+                case "topic":
+                    dataType = PrimitiveType.STRING;
+                    fieldValue = Encoding.ASCII.GetBytes(new string(reader.ReadChars(fieldLen - 6)));
+                    break;
+                default:
+                    throw new Exception($"{fieldName} not defined in ROSbag-format");
             }
+            return new FieldValue(fieldName, dataType, fieldValue);
+        }
+
+        /// <summary>
+        /// Reads a field name
```

Comment:
Not very descriptive summary, rather `reads until (and including) first "=", returning what was read, discarding "="`